packages/@ember/-internals/glimmer/lib/helper.ts

```diff
@@ -30,16 +39,10 @@ export interface SimpleHelper<T = unknown> {
   compute: HelperFunction<T>;
 }
 
-export function isHelperFactory(
-  helper: any | undefined | null
-): helper is Factory<SimpleHelper | HelperInstance, HelperFactory<SimpleHelper | HelperInstance>> {
-  return (
-    typeof helper === 'object' && helper !== null && helper.class && helper.class.isHelperFactory
-  );
-}
+const INTERNAL_MANAGERS = new WeakSet();
```

Comment:
I'm not sure it matters much, but I originally didn't use a WeakSet because the only type of manager it was important to pass the `factory` for is `ClassicHelperManager`. That manager is stateless, and a `===` comparison for that manager is going to be faster even that checking if the value exists in a `WeakSet`.

Reply:
I think we still need it actually, for the feature flagging. We only want to use the `customHelper` method for our own internal helper managers when the feature flag is disabled.